rmw_connext_cpp/src/rmw_take.cpp

```diff
@@ -195,6 +202,137 @@ _take(
+    return RMW_RET_ERROR;
+  }
+
+  if (count > message_sequence->capacity) {
+    RMW_SET_ERROR_MSG("insufficient capacity in message sequence");
+    return RMW_RET_ERROR;
+  }
+
+  if (count > message_info_sequence->capacity) {
+    RMW_SET_ERROR_MSG("insufficient capacity in message info sequence");
+    return RMW_RET_ERROR;
+  }
+
+  if (!message_info_sequence) {
+    RMW_SET_ERROR_MSG("message info sequence handle is null");
+    return RMW_RET_ERROR;
+  }
+
+  if (!taken) {
+    RMW_SET_ERROR_MSG("taken handle is null");
+    return RMW_RET_ERROR;
+  }
+
+  ConnextStaticSubscriberInfo * subscriber_info =
+    static_cast<ConnextStaticSubscriberInfo *>(subscription->data);
+  if (!subscriber_info) {
+    RMW_SET_ERROR_MSG("subscriber info handle is null");
+    return RMW_RET_ERROR;
+  }
+  DDS::DataReader * topic_reader = subscriber_info->topic_reader_;
+  if (!topic_reader) {
+    RMW_SET_ERROR_MSG("topic reader handle is null");
+    return RMW_RET_ERROR;
+  }
+  const message_type_support_callbacks_t * callbacks = subscriber_info->callbacks_;
+  if (!callbacks) {
+    RMW_SET_ERROR_MSG("callbacks handle is null");
+    return RMW_RET_ERROR;
+  }
+
+  DDS::DataReader * dds_data_reader = topic_reader;
+  bool ignore_local_publications = subscription->options.ignore_local_publications;
+
+  ConnextStaticSerializedDataDataReader * data_reader =
+    ConnextStaticSerializedDataDataReader::narrow(topic_reader);
+  if (!data_reader) {
+    RMW_SET_ERROR_MSG("failed to narrow data reader");
+    return false;
+  }
+
+  ConnextStaticSerializedDataSeq dds_messages;
+  DDS::SampleInfoSeq sample_infos;
+
+  *taken = 0;
+
+  DDS::ReturnCode_t status = data_reader->take(
+    dds_messages,
+    sample_infos,
+    count,
+    DDS::ANY_SAMPLE_STATE,
+    DDS::ANY_VIEW_STATE,
+    DDS::ANY_INSTANCE_STATE);
+
+  if (status == DDS::RETCODE_NO_DATA) {
+    data_reader->return_loan(dds_messages, sample_infos);
+    return RMW_RET_OK;
+  }
+  if (status != DDS::RETCODE_OK) {
+    data_reader->return_loan(dds_messages, sample_infos);
+    RMW_SET_ERROR_MSG("take failed");
+    return RMW_RET_ERROR;
+  }
+
+  for (int ii = 0; ii < dds_messages.length(); ++ii) {
+    bool ignore_sample = false;
+    const DDS::SampleInfo & sample_info = sample_infos[ii];
+    if (!sample_info.valid_data) {
+      ignore_sample = true;
+    } else if (ignore_local_publications && is_local_publication(sample_info, dds_data_reader)) {
+      ignore_sample = true;
+    }
+
+    if (!ignore_sample) {
+      rcutils_uint8_array_t cdr_stream = rcutils_get_zero_initialized_uint8_array();
+
+      cdr_stream.buffer_length = dds_messages[ii].serialized_data.length();
+      cdr_stream.buffer_capacity = dds_messages[ii].serialized_data.length();
+      cdr_stream.buffer = reinterpret_cast<uint8_t *>(&dds_messages[ii].serialized_data[0]);
+
+      if (callbacks->to_message(
+          &cdr_stream,
+          message_sequence->data[*taken]))
```

Comment:
nitpick: I'd do it like this:

```c++
      if (callbacks->to_message(&cdr_stream, message_sequence->data[*taken]))
      {
        rmw_gid_t * sender_gid = &message_info_sequence->data[*taken].publisher_gid;
        sender_gid->implementation_identifier = rti_connext_identifier;
        memset(sender_gid->data, 0, RMW_GID_STORAGE_SIZE);
        auto detail = reinterpret_cast<ConnextPublisherGID *>(sender_gid->data);
        detail->publication_handle = sample_info.publication_handle;
        (*taken)++;
      }
```

I don't think it's long enough to need wrapping.

Reply:
You are correct, fixed.